### ISSUE TITLE
Assertion failed on pointer delete[] while running in Debug mode

### ISSUE THREAD

**Issue by leandor** [wrye-bash/CBash#16]
Error:
![image](https://cloud.githubusercontent.com/assets/1095373/19668591/3529e334-9a2f-11e6-8013-071029c3e3af.png)

[Code](https://github.com/wrye-bash/CBash/blob/4386546632cb18d8acd512f59ff21bf366e5fe4a/src/Common.cpp#L1077)

``` c++
NonNullStringRecord::~NonNullStringRecord()
    {
    if(DiskSize == 0)
        delete []_value;  // <<<< HERE
    }
```

Probably caused by double delete of that pointer.

**Comment by leandor** [comment]
Well, it took me some time, but I finally found the reason for that assert.  (And I'm glad I did, as I didn't want to do big changes before finishing this issue)

It was because sometimes there are records that should contain a SCXI field, but the size of this field it's zero. And the code wasn't taking into account that condition, and since the condition for freeing its internal value buffer was that the Stored Size (meaning, the size of the element in the shared buffer) were 0, at the end it was trying to free a pointer that didn't point to the actual HEAP, but instead it was a sub-pointer to the shared buffer memory (filemap).
